lib/logstash/pipeline.rb

```diff
@@ -252,6 +252,7 @@ def outputworker
   #
   # This method is intended to be called from another thread
   def shutdown
+    Thread.new { report_inflight_events(@input_to_filter, @filter_to_output, @outputs) }
```

Comment:
Can we encapsulate this into a separate method somehow? Maybe `report_inflight_events_async(....)` or instead of Thread.new, use Concurrent::Future? I dunno, feels weird to create a thread like this.